File: leetcode/python/0011_container_with_most_water.py
class Solution:
  def maxArea(self, height: List[int]) -> int:
    # loop through height using two pointers i, j from (0, len(s)), (len(s) - 1, 0) respectively
    #   area = (j - i) * min(height[i], height[j])
    #   res = max(res, area)
    #   if height[j] > height[i]: i += 1
    #   else: j -= 1
    i, j = 0, len(height) - 1
    res = 0
    
    while i < j:
      area = (j - i) * min(height[i], height[j])
      res = max(res, area)

      if height[j] > height[i]: i += 1
      else: j -= 1
    
    return res

File: leetcode/cpp/0011_container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int area = 0;

        while (l < r)
        {
            // Calculating the max area
            area = max(area, min(height[l], height[r]) * (r - l));

            if (height[l] < height[r])
                l += 1;

            else
                r -= 1;
        }
        
        return area;
    }
};